src/include/duckdb.h

```diff
@@ -1328,6 +1350,76 @@ Test a `duckdb_timestamp` to see if it is a finite value.
 */
 DUCKDB_API bool duckdb_is_finite_timestamp(duckdb_timestamp ts);
 
+/*!
+Decompose a `duckdb_timestamp_s` object into a `duckdb_timestamp_struct`.
+
+* @return The `duckdb_timestamp_struct` with the decomposed elements.
+*/
+DUCKDB_API duckdb_timestamp_struct duckdb_from_timestamp_s(duckdb_timestamp_s ts_s);
+
+/*!
+Re-compose a `duckdb_timestamp_s` from a duckdb_timestamp_struct.
+
+* @param ts The de-composed elements in a `duckdb_timestamp_struct`.
+* @return The `duckdb_timestamp_s` element.
+*/
+DUCKDB_API duckdb_timestamp_s duckdb_to_timestamp_s(duckdb_timestamp_struct ts);
```

Comment:
I think all of these conversion functions to and from the struct are a bit much - in general we're not really using the `duckdb_timestamp_struct` anymore, and it's trivial to convert between `duckdb_timestamp_s` and `duckdb_timestamp` manually anyway if you need to. Perhaps we can leave these out?